inc/dashboard/namespace.php

```diff
@@ -0,0 +1,436 @@
+ */
+function render_views() {
+	global $post;
+	$views = get_block_views( $post->post_name, [ 'days' => get_days_view() ] )['views'];
+	?>
+	<div class="post--views"><?php echo number_format_i18n( $views ); ?></div>
+	<?php
+}
+
+/**
+ * Render the average conversion rate column data.
+ */
+function render_average_conversion_rate() {
+	global $post;
+	$block = get_block_views( $post->post_name, [ 'days' => get_days_view() ] );
+	$conversions = $block['conversions'];
+	$views = $block['views'];
+	$rate = round( calculate_average_conversion_rate( [
+		'conversions' => $conversions,
+		'views' => $views,
+	] ) * 100 );
+	?>
+	<div class="post--avg-conversion-rate"><?php echo absint( $rate ); ?>%
+	<?php
+}
+
+/**
+ * Return the days url query string.
+ *
+ * @return int The number of days to show insights for. Defaults to 7.
+ */
+function get_days_view() : int {
+	return isset( $_GET['days'] ) ? wp_unslash( absint( $_GET['days'] ) ) : 7;
+}
+
+/**
+ * Calculate average conversion rate.
+ *
+ * @param array $block_data An array of block data.
+ * @param string $block_id A single block clientID.
+ *
+ * @return float The conversion rate (calculated by conversions / views).
+ */
+function calculate_average_conversion_rate( array $block_data = [], string $block_id = '' ) : float {
+	if ( empty( $block_data ) ) {
+		$block_data = [
+			'conversions' => 0,
+			'views' => 0,
+		];
+
+		if ( ! empty( $block_id ) ) {
+			$block_data = get_block_data( $block_id );
+		}
+	}
+
+	// Avoid division by zero.
+	if ( $block_data['views'] === 0 ) {
+		return 0;
+	}
+
+	return $block_data['conversions'] / $block_data['views'];
+}
+
+/**
+ * Aggregate and map the data from the ES query.
+ *
+ * @param array $buckets The ElasticSearch query data.
+ *
+ * @return array An array of aggregated data.
+ */
+function get_aggregate_data( array $buckets ) : array {
+	$data = [];
+
+	foreach ( $buckets as $block ) {
+		$block_id = $block['key'];
+		$block_data = wp_list_pluck( $block['events']['buckets'], 'doc_count', 'key' );
+		$views = $block_data['experienceView'] ?? 0;
+		$conversions = $block_data['conversion'] ?? 0;
+		$data[ $block_id ]['views'] = $views;
+		$data[ $block_id ]['conversions'] = $conversions;
+		$data[ $block_id ]['avg_conversion_rate'] = calculate_average_conversion_rate( [
+			'views' => $views,
+			'conversions' => $conversions,
+		] );
+	}
+
+	return $data;
+}
+
+/**
+ * Sort data by conversion rate.
+ *
+ * @param array $list The array of analytics data by block id.
+ * @param string $order The order to sort by. Accepted values are 'asc' or 'desc'.
+ *
+ * @return array The sorted array.
+ */
+function sort_by_conversion_rate( array $list, string $order = 'desc' ) : array {
+	// If an invalid value was passed to $order, default to 'desc'.
+	if ( ! in_array( $order, [ 'asc', 'desc' ], true ) ) {
+		$order = 'desc';
+	}
+
+	$avg_conversion_rate = array_column( $list, 'avg_conversion_rate' );
+	$sort_order = 'SORT_' . strtoupper( $order );
```

Comment:
The sort flags are constants not strings so I'm not sure this will work. Use something like:
```suggestion
	$sort_order = ( $order === 'desc' ) ? SORT_DESC : SORT_ASC;
```